### DISCUSSION TITLE
how to use non-maximum suppression

### DISCUSSION THREAD

**Discussion by moziada** [facebookresearch/detectron2#2921] [Q&A]
i am trying to use MODEL.ROI_HEADS.NMS_THRESH_TEST to remove all the boxes that have IOU larger than 0.1 but there are an overleaping boxes on the bus and refuses to suppress any one of them 
```
cfg = get_cfg()
cfg.merge_from_file(model_zoo.get_config_file("COCO-InstanceSegmentation/mask_rcnn_X_101_32x8d_FPN_3x.yaml"))
cfg.MODEL.ROI_HEADS.NMS_THRESH_TEST = 0.1
cfg.MODEL.WEIGHTS = 'models/model_final_2d9806.pkl'
pred = DefaultPredictor(cfg)
outputs = pred(img)
```
![1533130050-cairo_main_bboxes3](https://user-images.githubusercontent.com/60968402/115029960-d473bc80-9ec6-11eb-8e66-1ec874ebecb8.jpg)

i found that one of the boxes belongs to class 5 with score 0.98, and the other belongs to class 7 with score 0.26

**Comment by mhrrs** [comment]
Hey did you ever figure this out? I have started to look into it in the post-processing of predictions from the DefaultPredictor and still haven't found any great answers.
